test/functional/wallet_miniscript_decaying_multisig_descriptor_psbt.py

```diff
@@ -0,0 +1,137 @@
+        multisig = self.create_multisig(external_xpubs, internal_xpubs)
+
+        self.log.info("Get a mature utxo to send to the multisig...")
+        coordinator_wallet = self.node.get_wallet_rpc(self.node.createwallet(wallet_name=f"coordinator", descriptors=True)["name"])
+        self.generatetoaddress(self.node, 101, coordinator_wallet.getnewaddress())
+
+        self.log.info("Send funds to the multisig's receiving address...")
+        deposit_amount = 6.15
+        coordinator_wallet.sendtoaddress(multisig.getnewaddress(), deposit_amount)
+        self.generate(self.node, 1)
+        assert_approx(multisig.getbalance(), deposit_amount, vspan=0.001)
+
+        self.log.info("Send transactions from the multisig as required signers decay...")
+        sent = 0
+        for locktime in [0] + self.locktimes:
+            self.log.info(f"At block height >= {locktime} this multisig is {self.M}-of-{self.N}")
+            current_height = self.node.getblock(self.node.getbestblockhash())['height']
+
+            amount = 1.5
+            receiver = signers[0]
+
+            # in this test each signer signs the same psbt "in series" one after the other.
+            # Another option is for each signer to sign the original psbt, and then combine
+            # and finalize these. In some cases this may be more optimal for coordination.
+            psbt = multisig.walletcreatefundedpsbt(inputs=[], outputs={receiver.getnewaddress(): amount}, feeRate=0.00010, locktime=locktime)
+            # the random sample asserts that any of the signing keys can sign for the 3-of-4,
+            # 2-of-4, and 1-of-4. While this is basic behavior of the miniscript thresh primitive,
+            # it is a critical property of this wallet.
+            for i, m in enumerate(random.sample(range(self.M), self.M)):
+                psbt = signers[m].walletprocesspsbt(psbt["psbt"])
+                assert_equal(psbt["complete"], i == self.M - 1)
+
+            if self.M < self.N:
+                self.log.info(f"Check that the time-locked transaction is too immature to spend with {self.M}-of-{self.N} at block height {current_height}...")
+                assert_equal(current_height >= locktime, False)
+                assert_raises_rpc_error(-26, "non-final", multisig.sendrawtransaction, psbt["hex"])
+
+                self.log.info(f"Generate blocks to reach the time-lock block height {locktime} and broadcast the transaction...")
+                self.generate(self.node, locktime - current_height)
+            else:
+                self.log.info("We can always spend with all signers!")
+
+            multisig.sendrawtransaction(psbt["hex"])
+            sent += amount
+
+            self.log.info("Check that balances are correct after the transaction has been included in a block...")
+            self.generate(self.node, 1)
+            assert_approx(multisig.getbalance(), deposit_amount - sent, vspan=0.001)
+            assert_equal(receiver.getbalance(), sent)
```

Comment:
because the receiver gets the full amount sent, while the sender sent the full amount plus a small txn fee (which is why sender needs to use assert approx to keep it simple)